Sources/MongoSwift/MongoCursor.swift

```diff
@@ -67,11 +78,16 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
         self.cursorType = cursorType ?? .nonTailable
         self.decoder = decoder
         self.error = nil
-
+        self.cached = .none
+        // If there was an error constructing the cursor, throw it.
         if let error = self.getMongocError() {
             self.close()
             throw error
         }
+
+        // Cache the first document.
```

Comment:
so initially I had planned to put `cacheDocument` in its own method. however, the only time we ever needed to call it was from right here, and there is no harm in always caching the first document. as I alluded to in our conversation earlier I think having it as a standalone method complicates the different states the driver can end up in.